### ISSUE TITLE
[desk-tool] It should be possible to unset the values from block editor

### ISSUE THREAD

**Issue by kmelve** [sanity-io/sanity#931]
It seems that the block editor will leave an array with a `span` object with an empty `text`. It should be possible to unset the array-field in the GUI, or else it's kinda cumbersome to check for “empty” block content in frontends. 

Example of value from an “empty” block editor: 

```json
"subtitle": [
        {
          "_type": "block",
          "_key": "7a7ace034645",
          "style": "normal",
          "markDefs": [],
          "children": [
            {
              "_type": "span",
              "_key": "7a7ace0346450",
              "text": "",
              "marks": []
            }
          ]
        }
      ]
```

**Comment by alevroub** [comment]
#706 👀

**Comment by davesag** [comment]
I'd call this a bug.  If I remove all content from a BlockContent field I'd expect to be able to then query `!defined(subtitle)` or, in a custom validation, check for `valueOfField('subtitle') === undefined` rather than something like

```
const isBlockEmpty = content => !content || !(content[0] || {}).(children[0] || {}).text
```

**Comment by rexxars** [comment]
Solved in v0.140.9

**Comment by mornir** [comment]
Ah! This is actually the problem I reported here: https://github.com/sanity-io/sanity/issues/1827
It is indeed possible to unset the values in the block editor. However, since marked texts have been removed, I think it's not reasonable to expect from content editors that they remove marks from the empty text, especially the marked text is only one empty space, like I reported.

**Wouldn't it be possible to automatically unset the values when the corresponding marked text is removed?**

Currently, I still have to write a not very elegant validation rule in the front end, like @davesag does. And there are no warnings against this issue in portable-text libraries.
